datus-snowflake/datus_snowflake/connector.py

```
@@ -0,0 +1,884 @@
+
+            if schema_name:
+                query_tables = query_tables.filter(pc.equal(query_tables["schema_name"], schema_name))
+            else:
+                query_tables = query_tables.filter(
+                    pc.invert(pc.is_in(query_tables["schema_name"], pa.array(self._sys_schemas(), type=pa.string())))
+                )
+
+            if tables:
+                query_tables = query_tables.filter(pc.is_in(query_tables["name"], pa.array(tables, type=pa.string())))
+
+            return query_tables
```

Ruff (0.14.3)

Consider moving this statement to an `else` block

(TRY300)